repos.bzl

```diff
@@ -2639,3 +1861,123 @@ def go_repositories():
         sum = "h1:9zdDQZ7Thm29KFXgAX/+yaf3eVbP7djjWp/dXAppNCc=",
         version = "v0.0.0-20190717185122-a985d3407aa7",
     )
+    go_repository(
+        name = "com_github_burntsushi_xgb",
+        build_file_generation = "on",
+        build_file_proto_mode = "disable",
+        importpath = "github.com/BurntSushi/xgb",
+        sum = "h1:1BDTz0u9nC3//pOCMdNH+CiXJVYJh5UQNCOBG7jbELc=",
+        version = "v0.0.0-20160522181843-27f122750802",
+    )
+    go_repository(
+        name = "com_github_googlecloudplatform_testgrid",
+        build_file_generation = "on",
+        build_file_proto_mode = "disable",
```

Comment:
This is the problematic line. We aren't generating build files for protobufs, which is causing issues in dependency resolution inside the bazel sandbox. It looks like this is being set by the repo infra bits to update dependencies.

@fejta Thoughts on this?

Reply:
We aren't generating BUILD files for protobufs, but that shouldn't be an issue, since the generated protobuf Go code is checked in: https://github.com/GoogleCloudPlatform/testgrid/blob/v0.0.1-alpha.3/pb/config/config.pb.go

I'm not sure why there is a missing Bazel dependency on `github.com/GoogleCloudPlatform/testgrid/pb/custom_evaluator` (used here: https://github.com/GoogleCloudPlatform/testgrid/blob/v0.0.1-alpha.3/pb/config/config.pb.go#L24).

Reply:
... though setting this to `default` does work. I wonder if the BUILD files that are being autogenerated are wrong somehow.

Reply:
these lines seem somewhat suspicious:
```
DEBUG: /root/.cache/bazel/_bazel_root/92c657af61c81ec6a823dc4627a93ca5/external/bazel_gazelle/internal/go_repository.bzl:182:13: gazelle: gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/state" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto and @com_github_googlecloudplatform_testgrid//pb/state:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/summary" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/summary:summary_go_proto and @com_github_googlecloudplatform_testgrid//pb/summary:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/state" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto and @com_github_googlecloudplatform_testgrid//pb/state:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/summarizer:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/summary" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/summary:summary_go_proto and @com_github_googlecloudplatform_testgrid//pb/summary:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/updater:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/updater:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/state" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto and @com_github_googlecloudplatform_testgrid//pb/state:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//cmd/updater:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/state" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto and @com_github_googlecloudplatform_testgrid//pb/state:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//config/yamlcfg:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//config/yamlcfg:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//config:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//config:go_default_test imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/config:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/custom_evaluator" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/custom_evaluator:custom_evaluator_go_proto and @com_github_googlecloudplatform_testgrid//pb/custom_evaluator:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/custom_evaluator:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/test_status" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/test_status:test_status_go_proto and @com_github_googlecloudplatform_testgrid//pb/test_status:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/response:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/response:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/summary" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/summary:summary_go_proto and @com_github_googlecloudplatform_testgrid//pb/summary:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/state:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/updater:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/config" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto and @com_github_googlecloudplatform_testgrid//pb/config:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/updater:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/state" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto and @com_github_googlecloudplatform_testgrid//pb/state:go_default_library. # gazelle:resolve may be used to disambiguate
gazelle: rule @com_github_googlecloudplatform_testgrid//pb/updater:go_default_library imports "github.com/GoogleCloudPlatform/testgrid/pb/summary" which matches multiple rules: @com_github_googlecloudplatform_testgrid//pb/summary:summary_go_proto and @com_github_googlecloudplatform_testgrid//pb/summary:go_default_library. # gazelle:resolve may be used to disambiguate
```

Reply:
Maybe try adding this line? https://github.com/kubernetes/test-infra/blob/1f3ba2f25ec728cf73398d6be1b867f638e7f120/BUILD.bazel#L2-L5

Reply:
> Maybe try adding this line? https://github.com/kubernetes/test-infra/blob/1f3ba2f25ec728cf73398d6be1b867f638e7f120/BUILD.bazel#L2-L5

Does not change anything over here.

Reply:
Try a gazelle:resolve? https://github.com/bazelbuild/bazel-gazelle#dependency-resolution

Reply:
Tried to set `gazelle:resolve` directives, didn't help either, `bazel test //...` still fails.

<details><summary><tt>gazelle:resolve</tt> directives</summary>

```
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/config            @com_github_googlecloudplatform_testgrid//pb/config:config_go_proto
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/custom_evaluator  @com_github_googlecloudplatform_testgrid//pb/custom_evaluator:custom_evaluator_go_proto
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/state             @com_github_googlecloudplatform_testgrid//pb/state:state_go_proto
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/summary           @com_github_googlecloudplatform_testgrid//pb/summary:summary_go_proto
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/test_status       @com_github_googlecloudplatform_testgrid//pb/test_status:test_status_go_proto
```
or
```
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/config            @com_github_googlecloudplatform_testgrid//pb/config:go_default_library
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/custom_evaluator  @com_github_googlecloudplatform_testgrid//pb/custom_evaluator:go_default_library
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/state             @com_github_googlecloudplatform_testgrid//pb/state:go_default_library
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/summary           @com_github_googlecloudplatform_testgrid//pb/summary:go_default_library
# gazelle:resolve go github.com/GoogleCloudPlatform/testgrid/pb/test_status       @com_github_googlecloudplatform_testgrid//pb/test_status:go_default_library
```

</details>

Reply:
seems to work with https://github.com/kubernetes/release/pull/930#issuecomment-572658606